tests/unit-tests/platforms/android/server/test_display.cpp

```diff
@@ -1165,3 +1165,71 @@ TEST_F(Display, does_not_invalidate_display_buffers_when_it_promised_not_to)
         EXPECT_THAT(db->transformation(), AnyOf(Eq(rotate_inverted), Eq(rotate_none)));
     }
 }
+
+TEST_F(Display, does_invalidate_display_buffers_when_it_promised_to)
+{
+    using namespace testing;
+    std::function<void()> hotplug_fn = []{};
+    bool external_connected = false;
+    stub_db_factory->with_next_config([&](mtd::MockHwcConfiguration& mock_config)
+    {
+        ON_CALL(mock_config, active_config_for(mga::DisplayName::primary))
+            .WillByDefault(Return(mtd::StubDisplayConfigurationOutput{
+                primary_output_id, {20,20}, {4,4}, mir_pixel_format_abgr_8888, 50.0f, true}));
+        ON_CALL(mock_config, active_config_for(mga::DisplayName::external))
+            .WillByDefault(Invoke([&](mga::DisplayName)
+            {
+                return mtd::StubDisplayConfigurationOutput{external_output_id,
+                    {20,20}, {4,4}, mir_pixel_format_abgr_8888, 50.0f, external_connected};
+            }));
+        EXPECT_CALL(mock_config, subscribe_to_config_changes(_,_))
+            .WillOnce(DoAll(SaveArg<0>(&hotplug_fn), Return(std::make_shared<char>('2'))));
+    });
+
+    mga::Display display(
+        stub_db_factory,
+        stub_gl_program_factory,
+        stub_gl_config,
+        null_display_report,
+        null_anw_report,
+        mga::OverlayOptimization::enabled);
+
+    std::vector<mg::DisplayBuffer*> active_dbs;
+
+    display.for_each_display_sync_group(
+        [&active_dbs](auto& sync_group)
+        {
+            sync_group.for_each_display_buffer(
+                [&active_dbs](auto& db)
+                {
+                    active_dbs.push_back(&db);
+                });
+        });
```

Comment:
Also remove this (see the other comment).

Reply:
fixed